src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -126,6 +227,60 @@ public String getPrefixName()
         }
     }
 
+    private static void fillInBytes(
+        byte[] addressBytes,
+        int index,
+        String ipv6Group)
+    {
+        short res = parseShort(ipv6Group, 16);
+        addressBytes[index++] = (byte) ((res >> 8) & 0xff);
+        addressBytes[index] = (byte) (res & 0xff);
+    }
+
+    private static void fillInIpv6HexCompressed(
+        String address,
+        byte[] addressBytes)
+    {
+        final Matcher ipv6Matcher = IPV6_HEX_COMPRESSED_MATCHER.get().reset(address);
+        ipv6Matcher.matches();
+        int endIndex = 7;
+        Arrays.fill(addressBytes, (byte) 0);
+        for (int i = 0; i < 7; i++)
+        {
+            String ipv6Group = ipv6Matcher.group(i + 1);
+            if (ipv6Group == null)
+            {
+                break;
+            }
+            fillInBytes(addressBytes, 2 * i, ipv6Group);
+        }
+        for (int i = 14; i > 7; i--)
```

Comment:
Same here, use `group` instead of `i` and use the actual group numbers, so `15` down to `8`.

Reply:
done replace all `i`